packages/@extension-kid/toast-notifications/ToastNotification.ts

```diff
@@ -18,30 +18,40 @@ type ToastCallback = (
   notification: ToastNotification
 ) => void;
 
+interface ToastTranslatableText {
+  id: string;
+  values?: Record<string, string>;
+}
+
 interface ToastNotificationOptions {
   appearance?: ToastAppearance;
   autodismiss?: boolean;
-  description?: string;
+  description?: string | ToastTranslatableText;
   callback?: ToastCallback;
-  primaryActionText?: string;
-  secondaryActionText?: string;
+  primaryActionText?: string | ToastTranslatableText;
+  secondaryActionText?: string | ToastTranslatableText;
 }
 
 class ToastNotification extends Notification {
   static NotificationType = Symbol("Toast");
-  readonly title: string;
-  readonly description: string;
+  readonly title: string | ToastTranslatableText;
+  readonly description?: string | ToastTranslatableText;
   readonly appearance: ToastAppearance;
   readonly autodismiss: boolean;
   readonly callback: ToastCallback | undefined;
-  readonly primaryActionText: string | undefined;
-  readonly secondaryActionText: string | undefined;
+  readonly primaryActionText?: string | ToastTranslatableText;
+  readonly secondaryActionText?: string | ToastTranslatableText;
 
-  constructor(title: string, options: ToastNotificationOptions = {}) {
-    const description = options.description || "";
-    super(ToastNotification.NotificationType, `${title}|${description}`);
+  constructor(
+    title: string | ToastTranslatableText,
+    options: ToastNotificationOptions = {}
+  ) {
+    super(
+      ToastNotification.NotificationType,
+      `${title}|${options.description || ""}`
```

Comment:
If `title` is an object (`ToastTranslatableText`) I think this will be kinda wonky?

Reply:
Good catch, I just refactor that. I will fix this.

Reply:
Is the PR still in progress then?

Reply:
no, that was just one of the last things I completed and I missed the implication on this call.